arangod/IResearch/IResearchLinkMeta.cpp

```diff
@@ -398,6 +397,67 @@ bool FieldMeta::init(
       }
     }
   }
+
+#ifdef USE_ENTERPRISE
+  {
+    // optional string map<name, overrides>
+    constexpr std::string_view kFieldName{"nested"};
+    if (slice.hasKey(kFieldName)) {
+      auto field = slice.get(kFieldName);
+
+      if (!field.isObject()) {
+        errorField = kFieldName;
+
+        return false;
+      }
+
+      auto subDefaults = *this;
+
+      _nested.clear();  // reset to match either defaults or read values exactly
+
+      for (velocypack::ObjectIterator itr(field); itr.valid(); ++itr) {
+        auto key = itr.key();
+        auto value = itr.value();
+
+        if (!key.isString()) {
+          errorField = std::string{kFieldName} + "[" +
+                       basics::StringUtils::itoa(itr.index()) + "]";
+
+          return false;
+        }
+
+        auto name = key.copyString();
+
+        if (!value.isObject()) {
+          errorField = std::string{kFieldName} + "." + name;
+
+          return false;
+        }
+
+        std::string childErrorField;
+
+        if (!_nested[name]->init(server, value, childErrorField, defaultVocbase,
+                                 version, subDefaults, referencedAnalyzers,
+                                 nullptr)) {
+          errorField =
+              std::string{kFieldName} + "." + name + "." + childErrorField;
+
+          return false;
+        }
+      }
+    }
+  }
+  _hasNested = !_nested.empty();
+  if (!_hasNested) {
+    for (auto const& f : _fields) {
+      if (!f.value()->_nested.empty()) {
+        _hasNested = true;
+        break;
+      }
+    }
+  }
```

Comment:
maybe instead of iterate for childs we can put _hasNested from child to parent?
I not insist, because it's not important and it can be need more code